sdk/src/client/api/block_builder/transaction_builder/mod.rs

```diff
@@ -688,99 +690,108 @@ impl TransactionBuilder {
-                    // Insert after the output we need
-                    sorted_inputs.insert(position + 1, input);
                 }
-                None => {
-                    // insert before address
-                    let account_or_nft_address = match &input.output {
-                        Output::Account(account_output) => Some(Address::Account(AccountAddress::new(
-                            account_output.account_id_non_null(input.output_id()),
-                        ))),
-                        Output::Nft(nft_output) => Some(Address::Nft(NftAddress::new(
-                            nft_output.nft_id_non_null(input.output_id()),
-                        ))),
-                        _ => None,
-                    };
-
-                    if let Some(account_or_nft_address) = account_or_nft_address {
-                        // Check for existing outputs for this address, and insert before
-                        match sorted_inputs.iter().position(|input_signing_data| {
-                            let required_address = input_signing_data
-                                .output
-                                .required_address(commitment_slot_index, committable_age_range)
-                                // PANIC: safe to unwrap as non basic/alias/foundry/nft outputs are already filtered
-                                .unwrap()
-                                .expect("expiration unlockable outputs already filtered out");
-
-                            required_address == account_or_nft_address
-                        }) {
-                            Some(position) => {
-                                // Insert before the output with this address required for unlocking
-                                sorted_inputs.insert(position, input);
-                            }
-                            // just push output
-                            None => sorted_inputs.push(input),
-                        }
-                    } else {
-                        // just push basic or foundry output
-                        sorted_inputs.push(input);
+                Output::Nft(nft_output) => {
+                    for input in self
+                        .other
+                        .remove(&Address::Nft(NftAddress::new(
+                            nft_output.nft_id_non_null(input.borrow().output_id()),
+                        )))
+                        .into_iter()
+                        .flatten()
+                        .rev()
+                    {
+                        self.queue.push_front(input);
                     }
                 }
+                _ => (),
+            };
+            return Some(input);
+        }
+        if let Some(mut entry) = self.other.first_entry() {
+            if let Some(input) = entry.get_mut().pop_front() {
+                if entry.get().is_empty() {
+                    self.other.pop_first();
+                }
+                return Some(input);
```

Comment:
Can you please add some comments to the code what it's doing, bit hard to follow what this is for
Also some general thing like we had before `Inputs need to be sorted before signing, because the reference unlock conditions can only reference a lower index`

Reply:
Thanks